opencti-platform/opencti-graphql/src/database/middleware.js

```diff
@@ -565,7 +566,33 @@ const idLabel = (label) => {
   return isAnId(label) ? label : generateStandardId(ENTITY_TYPE_LABEL, { value: normalizeName(label) });
 };
 
+/**
+ * Verify that the Entity in createdBy is one of Identity entity.
+ * If not throw functional error to stop creation or update.
+ * @param context
+ * @param user
+ * @param createdById
+ * @returns {Bluebird.Promise<void>}
+ */
+export const validateCreatedBy = async (context, user, createdById) => {
+  if (createdById) {
+    const createdByEntity = await internalLoadById(context, user, createdById);
+    if (createdByEntity && createdByEntity.parent_types) {
+      if (!createdByEntity.parent_types.some((parent) => parent === ENTITY_TYPE_IDENTITY)) {
```

Comment:
I wonder if you could you use `isStixDomainObjectIdentity(createdByEntity.entity_type)` ?